docs/topics/native/native-app-with-c-and-libcurl.md

```diff
@@ -0,0 +1,260 @@
+2. Name new directory **nativeInterop/cinterop**. This is the default convention for header file locations, though it can
+   be overridden in the `build.gradle` file if you use a different location.
+3. Select this new subfolder and create a new `libcurl.def` file with **File | New | File**.
+4. Update your file with the following code:
+
+```c
+headers = curl/curl.h
+headerFilter = curl/*
+
+compilerOpts.linux = -I/usr/include -I/usr/include/x86_64-linux-gnu
+linkerOpts.osx = -L/opt/local/lib -L/usr/local/opt/curl/lib -lcurl
+linkerOpts.linux = -L/usr/lib/x86_64-linux-gnu -lcurl
+```
+
+* The first entry is `headers`, the list of header files to generate Kotlin stubs for. You can add multiple files to
+this entry, separating each with a `\` on a new line. In our case, it's only `curl.h`. The referenced files need to be
+relative to the folder where the definition file is or be available on the system path (in our case, it's `/usr/include/curl`).
+* The second line is the `headerFilter`. It shows what exactly is included. In C, all the headers are also included when
+one file references another one with the `#include` directive. Sometimes it's not necessary, and you can add this
+parameter [using glob patterns](https://en.wikipedia.org/wiki/Glob_(programming)) to fine-tune things.
+
+> `headerFilter` is an optional argument and is mostly used when the library is installed as a system library. You don't
+> want to fetch external dependencies (such as system `stdint.h` header) into the interop library. It may be important to
+> optimize the library size and fix potential conflicts between the system and the provided Kotlin/Native compilation
+> environment.
+>
+{type="note"}
+
+* The next lines are about providing linker and compiler options, which can vary depending on different target platforms.
+In our case, they are macOS (the `.osx` suffix) and Linux (the `.linux` suffix). Parameters without a suffix are also
+possible (for example, `linkerOpts=`) and applied to all platforms.
+
+The convention is that each library gets its definition file, usually with the same name as the library. For more
+information on all the options available to `cinterop`, see [the Interop section](native-c-interop.md).
+
+> You need to have the `curl` library binaries on your system to make the sample work. On macOS and Linux, it is usually
+> included. On Windows, you can build it from [sources](https://curl.haxx.se/download.html) (you'll need Visual Studio or
+> Windows SDK Commandline tools). For more details, see the [related blog post](https://jonnyzzz.com/blog/2018/10/29/kn-libcurl-windows/).
+> Alternatively, you may also want to consider a [MinGW/MSYS2](https://www.msys2.org/) `curl` binary.
+>
+{type="note"}
+
+## Add interoperability to the build process
+
+To use header files, let's make sure they are generated as a part of the build process. For this, add the following
```

Comment:
Omit "let's". This is a tutorial, that contains a list of actions that users should do.